projects/step-frontend/src/lib/modules/timeseries/metric-chart/metric-chart.component.ts

```diff
@@ -0,0 +1,232 @@
+import { AfterViewInit, Component, inject, Input, OnChanges, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { TSChartSeries, TSChartSettings } from '../chart/model/ts-chart-settings';
+import { TSTimeRange } from '../chart/model/ts-time-range';
+import {
+  BucketAttributes,
+  BucketResponse,
+  FetchBucketsRequest,
+  MetricAttribute,
+  MetricType,
+  TimeSeriesAPIResponse,
+  TimeSeriesService,
+} from '@exense/step-core';
+import { TimeSeriesUtils } from '../time-series-utils';
+import { UPlotUtils } from '../uplot/uPlot.utils';
+import { TimeSeriesConfig } from '../time-series.config';
+
+//@ts-ignore
+import uPlot = require('uplot');
+import { FilterUtils } from '../util/filter-utils';
+import { TimeseriesColorsPool } from '../util/timeseries-colors-pool';
+import MouseListener = uPlot.Cursor.MouseListener;
+import { TimeSeriesChartComponent } from '../chart/time-series-chart.component';
+
+type AggregationType = 'SUM' | 'AVG' | 'MAX' | 'MIN' | 'COUNT' | 'RATE' | 'MEDIAN' | 'PERCENTILE';
+
+interface MetricAttributeSelection extends MetricAttribute {
+  selected: boolean;
+}
+
+@Component({
+  selector: 'step-metric-chart',
+  templateUrl: './metric-chart.component.html',
+  styleUrls: ['./metric-chart.component.scss'],
+})
+export class MetricChartComponent implements OnInit, OnChanges {
+  chartSettings?: TSChartSettings;
+  @ViewChild(TimeSeriesChartComponent) chart!: TimeSeriesChartComponent;
+
+  @Input() filters: Record<string, any> = {};
+  @Input() settings!: MetricType;
+  @Input() range!: TSTimeRange;
+  @Input() allowGroupingChange = true;
+  @Input() allowMetricChange = true;
+  @Input() colorsPool: TimeseriesColorsPool = new TimeseriesColorsPool();
+
+  readonly AGGREGATES: AggregationType[] = ['SUM', 'AVG', 'MIN', 'MAX', 'COUNT', 'RATE', 'MEDIAN', 'PERCENTILE'];
+  selectedAggregate!: AggregationType;
+  groupingAttributes: MetricAttributeSelection[] = [];
+  selectedRange!: TSTimeRange; // when zooming, this will get smaller
```

Comment:
Should work like this:

```suggestion
  selectedRange = this.range;
```